### PR TITLE
Support ECR in Windows Environment

### PR DESCRIPTION
**Pull request by orangedeng**
Related issue https://github.com/rancher/rancher/issues/12018
Add a new catalog entry for Rancher ECR Windows.

### CODE REVIEW

File: infra-templates/ecr-windows/0/docker-compose.yml.tpl [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    io.rancher.container.pull_image: always
+    io.rancher.scheduler.affinity:host_label: io.rancher.host.os=windows
+    {{- if eq .Values.registry_in_which_environment "current" }}
+    io.rancher.container.create_agent: 'true'
+    io.rancher.container.agent.role: environment
+    io.rancher.container.agent.volumes_strategy: "skip"
+    io.rancher.container.create_agent_label: "no"
+    {{- end }}
+  tty: true
+  image: rancher/rancher-ecr-credentials-windows:v2.0.1
```
Comment (deniseschannon) [inline infra-templates/ecr-windows/0/docker-compose.yml.tpl]: Please update to:  `v3.0.0`

File: infra-templates/ecr-windows/0/rancher-compose.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+.catalog:
+  name: "ECR Credential Updater"
+  version: "v2.0.1"
```
Comment (deniseschannon) [inline infra-templates/ecr-windows/0/rancher-compose.yml]: Please update to:  `v3.0.0`

File: infra-templates/ecr-windows/0/rancher-compose.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  description: "Updates credentials for ECR in Rancher"
+  minimum_rancher_version: "v1.6.14"
```
Comment (deniseschannon) [inline infra-templates/ecr-windows/0/rancher-compose.yml]: Please update to `v1.6.15-rc1`

File: infra-templates/ecr-windows/config.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+name: ECR Credential Updater for Windows Environment
+description: |
+  Automatically updates AWS EC2 Container Registry credentials in Rancher.
+version: v2.0.1
```
Comment (deniseschannon) [inline infra-templates/ecr-windows/config.yml]: v3.0.0

**Review by deniseschannon — CHANGES_REQUESTED** [review]

**Comment by orangedeng** [comment]
@deniseschannon Updated

**Comment by deniseschannon** [comment]
LGTM

**Pull request merged**